### PR TITLE
Add support for env vars in configs. (#210)

### PR DESCRIPTION
**Pull request by JivanAmara**
Does a search and replace for $IDENTIFIER in config content and replaces with value from environment variables, or empty string if not set in environment.  After this sends the modified config content on for usual processing.

### CODE REVIEW

**Comment by ARolek** [comment]
@JivanAmara I made a few changes to the branch. Please take a look and let me know any questions you have. Let me handle the merge please so I can get the release notes and change log updated.

File: config/config.go [hunk 1]
```diff
@@ -138,25 +113,31 @@ func Parse(reader io.Reader, location string) (conf Config, err error) {
...
-		log.Printf("Problem reading from config reader: %v", err)
 		return nil, err
 	}
+
 	configStr := string(configBytes)
 
 	varPlaceHolders := varFinder.FindAllString(configStr, -1)
 	for _, ph := range varPlaceHolders {
 		// Get the environment variable value (drop the leading dollar sign ($))
 		envVal := os.Getenv(ph[1:])
```
Comment (gdey) [inline config/config.go]: Use `syscall.Getenv()` instead. It will tell you whether the variable exists or not.

Comment (ARolek) [reply config/config.go]: Good call. fixing coming up

File: config/config.go [hunk 2]
```diff
@@ -136,6 +114,39 @@ func Parse(reader io.Reader, location string) (conf Config, err error) {
 	return conf, err
 }
 
+// replaceEnvVars replaces environment variable placeholders in reader stream with values
+// i.e. "val = $VAR" -> "val = 3"
+func replaceEnvVars(reader io.Reader) (io.Reader, error) {
+	// Variable definition follows IEEE Std 1003.1-2001
+	//   A dollar sign ($) followed by an upper-case letter, followed by
+	//   zero or more upper-case letters, digits, or underscores (_).
+	regexStr := `\$[A-Z]+[A-Z1-9_]*`
```
Comment (ARolek) [inline config/config.go]: @gdey brought up another approach, using [os.Expand](https://golang.org/pkg/os/#Expand) which would be cross platform and cleaner than the regex approach.

Comment (gdey) [reply config/config.go]: Yeah, looking at that now. I would say go with the regex, but enhance the regex to deal with `${SHELL_NAME}` as well. Or We require that all shell var substitution is done with `${}` to make it more obvious. In which case we would still want the regex.  (For reference: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html )

**Pull request merged**